prometheus_blackbox_exporter/handlers/main.yaml

```
@@ -0,0 +1,16 @@
+# vim:ts=2:sw=2:et:ai:sts=2
+---
+- name: _blackbox_exporter__restart_blackbox_exporter
+  ansible.builtin.service:
+    name: prometheus-blackbox-exporter
+    state: restarted
+
+- name: _blackbox_exporter__reload_blackbox_exporter
+  ansible.builtin.service:
+    name: prometheus-blackbox-exporter
+    state: reloaded
+
+- name: _blackbox_exporter__reload_ferm
```

nitpick: I'd consider this a semi-public API, so you don't need to prefix the names (also, consistency with other roles)
```suggestion
- name: blackbox_exporter__restart_blackbox_exporter
  ansible.builtin.service:
    name: prometheus-blackbox-exporter
    state: restarted

- name: blackbox_exporter__reload_blackbox_exporter
  ansible.builtin.service:
    name: prometheus-blackbox-exporter
    state: reloaded

- name: blackbox_exporter__reload_ferm
```

---

Ah, I should have clarified with a comment in the PR description:

I used the underscore as a way to get away from the `name[casing]` lint, which wants all names to start with uppercase. (As you can see, I did capitalize task names, but `Blackbox_exporter__reload_X` just seemed too ugly.)

This means this module currently passes ‘production’. But if you think `name[casing]` will stay disabled forever, I will remove the underscore here and use lowercase task names?

(Or, well: even if it won’t stay disabled forever, we can opt for consistency, and fix everything at once down the line.)

---

ah, I see...

The problem here is that name is both a description and an indentifier... A while ago I prefixed all handlers to avoid unexpected clashes (even though that causes identical handlers to be executed twice), but never thought of them as names.

So there are a few issues:
* I want to keep `name[casing]` to have consistent task descriptions
* I want to avoid handler name clashes with unrelated roles; but clashes inside this repo/collection would be fine (and possibly an improvement)
* It would be nice if the handlers also have reasonable names.

So, we could either prefix with underscores as you did, or convert them into English phrases that include a namespace identifier (e.g. `Reload prometheus-blackbox-exporter (tina-roles)`), or leave things as they are and add lint exceptions for the handlers. WDYT?

---

0.  Agreed to enabling `name[casing]` down the line.

1.  I'd keep on using snake_case for handler names. In other projects I use things like `reload nginx` and `restart uwsgi`, but I find more than one unwieldy (wrt things like `git grep -w` or Vim's `*` operator).

2. A leading underscore seems best to me. As you wrote earlier, these can be considered "a semi-public API", so  a single leading underscore kind of signals that..?

    (If we go with this, I'd even consider dropping the doubled underscore: `_blackbox_exporter_reload_ferm`?)

3. (optional) Re:
    > clashes inside this repo/collection would be fine (and possibly an improvement)

    How would clashes work? Do we want to move to `_<action>_<service_name>`?

---

> 0. Agreed to enabling `name[casing]` down the line.

I thought it was already enabled

>     1. I'd keep on using snake_case for handler names. In other projects I use things like `reload nginx` and `restart uwsgi`, but I find more than one unwieldy (wrt things like `git grep -w` or Vim's `*` operator).

+1

>     2. A leading underscore seems best to me. As you wrote earlier, these can be considered "a semi-public API", so  a single leading underscore kind of signals that..?
>        (If we go with this, I'd even consider dropping the doubled underscore: `_blackbox_exporter_reload_ferm`?)

sounds good

>     3. (optional) Re:
>        > clashes inside this repo/collection would be fine (and possibly an improvement)

>        How would clashes work? Do we want to move to `_<action>_<service_name>`?

The issue is that ansible is terribly designed, and handlers have global scope once defined. So, if two completely unrelated roles define a handler with the same name, only the last one is used. This is a good thing when all the equally-named handlers are identical (so you don't relod ferm 20 times), but there is no way to prevent that when using 3rd party roles; so for that reason I started prefixing them like I do with variables (also global scope).

So, it would actually be good if the handlers in tina-roles reuse names, as long as there is a way to avoid clashes with 3rd party roles. A way to do that and keep the handler names readable is to name them something like "Reload <service> (tina-roles)" or something similar

---

We could do `_tina_reload_<service>`

---

SGTM!